src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,169 @@
+import core.basesyntax.exception.IllegalAgeException;
+import core.basesyntax.exception.InvalidLoginException;
+import core.basesyntax.exception.InvalidPasswordException;
+import core.basesyntax.exception.LoginExistingException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static final String NULL_TEST_FAILURE_MESSAGE = "Null fields "
+            + "should throw EmptyFieldException!";
+
+    private static final String LOGIN_EXISTS_FAILURE_MESSAGE = "Login already exists in storage "
+            + "should throw LoginExistingException!";
+    private static final String LOGIN_TEST_FAILURE_MESSAGE = "Login less than 6 letters "
+            + "should throw InvalidLoginException!";
+    private static final String PASSWORD_TEST_FAILURE_MESSAGE = "Password less than 6 letters "
+            + "should throw InvalidPasswordException!";
+    private static final String AGE_TEST_FAILURE_MESSAGE = "Age less than 18 "
+            + "or over 122 should throw IllegalAgeException!";
+    private static final String USER_BOB_LOGIN_DEFAULT = "bobjordan";
+    private static final String USER_ALICE_LOGIN_DEFAULT = "alicebaker";
+    private static final String LOGIN_WORSE_CASE = "bob";
+    private static final String USER_PASSWORD_DEFAULT = "password";
+    private static final String PASSWORD_WORSE_CASE = "pass";
+    private static final Integer USER_AGE_DEFAULT = 18;
+    private static final Integer LESS_AGE_WORSE_CASE = 16;
+    private static final Integer OVER_AGE_WORSE_CASE = 210;
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User bob;
+    private User alice;
+
+    @BeforeAll
+    public static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    public void registerUser_correctInputData() {
+        bob = new User(USER_BOB_LOGIN_DEFAULT, USER_PASSWORD_DEFAULT, USER_AGE_DEFAULT);
+        alice = new User(USER_ALICE_LOGIN_DEFAULT, USER_PASSWORD_DEFAULT, USER_AGE_DEFAULT);
+        registrationService.register(bob);
+        registrationService.register(alice);
+        User expectedFirstUser = storageDao.get(bob.getLogin());
+        User expectedSecondUser = storageDao.get(alice.getLogin());
+        assertEquals(expectedFirstUser, bob);
+        assertEquals(expectedSecondUser, alice);
+    }
+
+    @Test
+    public void registerDuplicateUser_throwsRuntimeException() {
+        bob = new User(USER_BOB_LOGIN_DEFAULT, USER_PASSWORD_DEFAULT, USER_AGE_DEFAULT);
+        alice = bob;
+        try {
+            registrationService.register(bob);
+            registrationService.register(alice);
+        } catch (LoginExistingException e) {
+            return;
+        }
+        fail(LOGIN_EXISTS_FAILURE_MESSAGE);

```

Comment:
[Common mistakes](https://github.com/proxychang/jv-registration-validation/blob/hw-jv-registration-validation/checklist.md#common-mistakes-jv-registration-validation)
- If you are expecting exception to be thrown in the test, you can do it this way:

`assertThrows(ExpectedException.class, () -> object.methodUnderTest(inputParams);`

This is a recommendation in common mistakes, so it's up to you if you would want to change it to make code cleaner.